packages/ui/internal/IdProvider/useId.js

```diff
@@ -0,0 +1,46 @@
+import useClass from '@enact/core/useClass';
+import React from 'react';
+
+import Provider from './Provider';
+
+/**
+ * Configuration for `useIdProvider`
+ *
+ * @typedef {Object} useIdProviderConfig
+ * @memberof ui/IdProvider
+ * @property {String} [prefix] Optional prefix for the identifier.
+ * @private
+ */
+
+/**
+ * Object returned by `useIdProvider`
+ *
+ * @typedef {Object} useIdProviderInterface
+ * @memberof ui/IdProvider
+ * @property {Function} [generateId] The function to generate id.
+ * @private
+ */
+
+/**
+ * Generate globally-unique identifiers
+ *
+ * @param {useIdProviderConfig} config Configuration options
+ * @returns {useIdProviderInterface}
+ * @private
+ */
+const useIdProvider = ({prefix}) => {
+	const provider = useClass(Provider, prefix);
+
+	React.useEffect(() => {
+		return () => provider.unload();
+	}, [provider]);
```

Comment:
Technically not required since `provider` doesn't change, but then lint will complain.  I'm not sure if there's any shortcut in the comparison logic if the array is empty.

Reply:
Agreed. Seemed easiest to follow the rule since a strict equality check on the object isn't an expensive operation.